Portage has just glide 3.x so use that and not glide 2.x.
This patch replaces the sed to configure in src_unpack in old ebuilds.
Index: vlc-0.8.5-test2/modules/video_output/glide.c
===================================================================
--- vlc-0.8.5-test2.orig/modules/video_output/glide.c
+++ vlc-0.8.5-test2/modules/video_output/glide.c
@@ -244,20 +244,11 @@ static void Display( vout_thread_t *p_vo
 
 static int OpenDisplay( vout_thread_t *p_vout )
 {
-    static char version[80];
-    GrHwConfiguration hwconfig;
     GrScreenResolution_t resolution = GR_RESOLUTION_800x600;
     GrLfbInfo_t p_front_buffer_info;                    /* front buffer info */
 
-    grGlideGetVersion( version );
     grGlideInit();
 
-    if( !grSstQueryHardware(&hwconfig) )
-    {
-        msg_Err( p_vout, "cannot get 3dfx hardware config" );
-        return( 1 );
-    }
-
     grSstSelect( 0 );
     if( !grSstWinOpen( 0, resolution, GR_REFRESH_60Hz,
                        GR_COLORFORMAT_ABGR, GR_ORIGIN_UPPER_LEFT, 2, 1 ) )
Index: vlc-0.8.5-test2/configure.ac
===================================================================
--- vlc-0.8.5-test2.orig/configure.ac
+++ vlc-0.8.5-test2/configure.ac
@@ -3938,8 +3938,8 @@ then
    CFLAGS="$CFLAGS -I/usr/include/glide"
    AC_CHECK_HEADER(glide.h,[   
       VLC_ADD_PLUGINS([glide])
-      VLC_ADD_LDFLAGS([glide],[-lglide2x -lm])
-      VLC_ADD_CPPFLAGS([glide],[-I/usr/include/glide])
+      VLC_ADD_LDFLAGS([glide],[-lglide3 -lm])
+      VLC_ADD_CPPFLAGS([glide],[-I/usr/include/glide3])
     ],[
       AC_MSG_ERROR([You don't have libglide. Install it or do not use --enable-glide])
      ])
